love/components/questions-form.tsx

```diff
@@ -59,75 +60,84 @@ export const QuestionsForm = (props: { questionType: QuestionType }) => {
   )
 }
 type loveAnswer = rowFor<'love_answers'>
-type loveAnswerState = Omit<loveAnswer, 'id' | 'created_time'>
+export type loveAnswerState = Omit<loveAnswer, 'id' | 'created_time'>
+
+const fetchPrevious = async (
+  id: number,
+  userId: string,
+  setForm: (
+    newState:
+      | loveAnswerState
+      | ((prevState: loveAnswerState) => loveAnswerState)
+  ) => void
+) => {
+  const res = await run(
+    db
+      .from('love_answers')
+      .select('*')
+      .eq('question_id', id)
+      .eq('creator_id', userId)
+  )
+  if (res.data.length) {
```

Comment:
Hmmm this is weird to pass in a setState as a parameter to a db fetching function like this, why not just set the state from the parent with the results of the function?